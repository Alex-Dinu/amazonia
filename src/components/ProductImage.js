import React from "react";
import PropTypes from "prop-types";

function ProductImage(props) {
  return (
    <div className="cart-image">
      <img src={props.imageSource} alt={props.altText}></img>
    </div>
  );
}

ProductImage.prototypes = {
  imageSource: PropTypes.string.isRequired,
  altText: PropTypes.string.isRequired,
};

export default ProductImage;
