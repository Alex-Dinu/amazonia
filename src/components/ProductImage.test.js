import React from "react";
import renderer from "react-test-renderer";
import ProductImage from "./ProductImage";

describe("The ProductImage component", () => {
  it("renders as expected", () => {
    const tree = renderer
      .create(
        <ProductImage
          imageSource={"imageSourcePath"}
          altText={"image alt text"}
        ></ProductImage>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
