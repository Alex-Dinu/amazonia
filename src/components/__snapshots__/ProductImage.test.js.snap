// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="cart-image"
>
  <img
    alt="image alt text"
    src="imageSourcePath"
  />
</div>
`;

exports[`The ProductImage component renders as expected 1`] = `
<div
  className="cart-image"
>
  <img
    alt="image alt text"
    src="imageSourcePath"
  />
</div>
`;
